Arrive alert

danielcwilson opened the pull request:
Vibrate and show a pop up when the user gets close to the chosen beacon

danielcwilson committed:
Show alert and vibrate when user is "near" destination beacon

Currently can only be used once before crashing, needs debugging. However for the one time it works its pretty good eh

danielcwilson committed:
Fix crash on second navigate attempt

Also remove an unnecessary log

danielcwilson committed:
Merge branch 'gradient-buttons-with-swiftlint' into arrive-alert

I love programming in like 3 different branches at once

danielcwilson committed:
Merge remote-tracking branch 'origin/dev' into arrive-alert

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -123,10 +124,22 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+                        self.present(alertController, animated: true, completion: nil)
+
+                        // clear circle and line
+                        self.lineShapeLayer.removeFromSuperlayer()
```
May check if `lineShapeLayer` and `circleShapeLayer `  are not null. I will check it myself if we need that.

danielcwilson replied:
They should never be nil since they're initialised at the same time as isNavigating being set to true

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -123,10 +124,22 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
                         isNavigating = false
-                        print("123223")
+                        // vibrate
+                        AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))
```
What sound system do you play? As I cannot hear anything from my phone.

danielcwilson replied:
Sound system? 
That line is just how iOS wants to make vibrations for whatever reason

henrycity replied:
Sound system as an alert: https://developer.apple.com/documentation/audiotoolbox/1405202-audioservicesplayalertsound

And I don't think we need that. If you look here: https://github.com/tvtri96/indoor-navigation/commit/eea3e93495c4587f38418d2f88fb3d425bf5d535. We don't need AudioServicesPlayAlertSound to show the alert.

danielcwilson replied:
What are you pointing out? 
I chose to add the vibration because I assume its not intended for the user to be staring at the phone, if theres just an alert box then they might miss their destination

henrycity replied:
I mean you don't need this line to show the alert `AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))`
This is enough
```
let alertController = UIAlertController(title: "Room Finder", message:
 +                            "You have arrived!", preferredStyle: UIAlertControllerStyle.alert)
 +                        alertController.addAction(UIAlertAction(title: "Dismiss", style: UIAlertActionStyle.default))
 +                        self.present(alertController, animated: true, completion: nil)

```

danielcwilson replied:
But then it doesn't vibrate? Or is that part of the alert controller?

henrycity replied:
Okay sorry my mistake. So everything is fine.

danielcwilson committed:
Fix index out of bounds error when calculating colour

Exact colours can be adjusted at another time with  group advice. 

Its very likely that I used the wrong condition so please test, I always mess things like that up 
Also I super disagree with lint i feel like if conditions should be surrounded with brackets but whatever

danielcwilson committed:
Less lines more better

tvtri96 commented:
Merged in feature/action-controller
